kernel/smp.c

```diff
@@ -93,19 +89,11 @@ void z_smp_init(void)
 {
 	(void)atomic_clear(&start_flag);
 
-#if defined(CONFIG_SMP) && CONFIG_MP_NUM_CPUS > 1
-	arch_start_cpu(1, _interrupt_stack1, CONFIG_ISR_STACK_SIZE,
-			smp_init_top, &start_flag);
-#endif
-
-#if defined(CONFIG_SMP) && CONFIG_MP_NUM_CPUS > 2
-	arch_start_cpu(2, _interrupt_stack2, CONFIG_ISR_STACK_SIZE,
-			smp_init_top, &start_flag);
-#endif
-
-#if defined(CONFIG_SMP) && CONFIG_MP_NUM_CPUS > 3
-	arch_start_cpu(3, _interrupt_stack3, CONFIG_ISR_STACK_SIZE,
-			smp_init_top, &start_flag);
+#if defined(CONFIG_SMP) && (CONFIG_MP_NUM_CPUS > 1)
```

Comment:
You don't actually need to test CONFIG_MP_NUM_CPUS in the #if, it's part of the loop conditional below.

Reply:
I had to put this here due to smp_init_top() not being defined otherwise.